### PR TITLE
Generics support

### PR DESCRIPTION
**Pull request by sgtsquiggs**
* Adds support for generics
* Moves `toType` util function to types package and makes it public
* Tests updated w/ generic cases
* Bumps golangci-lint in ci workflow to 1.54

### CODE REVIEW

File: internal/generator/method/method_test.go [hunk 1]
```diff
@@ -239,7 +262,8 @@ func TestNewMethod(t *testing.T) {
 			}
 
 			require.ElementsMatch(t, tt.want.ParameterNames, got.ParameterNames)
-			require.ElementsMatch(t, tt.want.ParametersNameAndType, got.ParametersNameAndType)
+			//require.ElementsMatch(t, tt.want.ParametersNameAndType, got.ParametersNameAndType)
```
Comment (dnisenbaum1) [inline internal/generator/method/method_test.go]: uncomment?

Comment (dnisenbaum1) [reply internal/generator/method/method_test.go]: or remove entirely

File: internal/loader/loader.go [hunk 2]
```diff
@@ -254,10 +269,19 @@ func loadFromStruct(f *ast.File, name string, info *types.Info) (*Result, error)
...
+			case *ast.StarExpr:
+				switch t := t.X.(type) {
+				case *ast.Ident:
+					ident = t
+				case *ast.IndexExpr:
+					// single type parameter
+					ident = t.X.(*ast.Ident)
+				case *ast.IndexListExpr:
+					// multiple type parameters
+					ident = t.X.(*ast.Ident)
```
Comment (dnisenbaum1) [inline internal/loader/loader.go:283]: is handling both `*ast.IndexListExp` and `*ast.IndexExpr` cases meant to be indentical?

File: internal/generator/method/method.go [hunk 3]
```diff
@@ -6,23 +6,19 @@ import (
...
-	Name() string
-}
-
 // Method holds all of the data for code generation on a specific method signature
 type Method struct {
 	Name                  string
 	HasContext            bool
 	ReturnsError          bool
 	HasVariadic           bool
+	GenericsNameAndType   []jen.Code
```
Comment (dnisenbaum1) [inline internal/generator/method/method.go]: is this used anywhere?

File: internal/loader/loader.go (hunk 2 above)
Comment (sgtsquiggs) [reply internal/loader/loader.go:283]: yep - one is analog to `MyType[T any]` and the other is `MyType[T any, U any, V any]`... single type arg vs multiple. I should have a new unit test that covers both.

File: internal/generator/method/method.go (hunk 3 above)
Comment (sgtsquiggs) [reply internal/generator/method/method.go]: nope, removed

File: internal/generator/method/method_test.go (hunk 1 above)
Comment (sgtsquiggs) [reply internal/generator/method/method_test.go]: updated

**Review by dnisenbaum1 — APPROVED** [review]

**Review by tzachshabtay — APPROVED** [review]

**Pull request merged**